### PR TITLE
paywall: update types for Lock and Key

### PR DESCRIPTION
**Pull request by cellog**
# Description

This updates the types for `Lock` and `Key` to reflect reality.

# Issues

Refs #2903

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: paywall/src/unlockTypes.ts [hunk 1]
```diff
@@ -102,6 +107,6 @@ export interface Key {
   transactions: Transaction[]
   status: string
   confirmations: number
-  owner: string
+  owner: string | null
```
Comment (cnasc) [inline paywall/src/unlockTypes.ts]: In which cases do keys not have owners?

File: paywall/src/unlockTypes.ts [hunk 2]
```diff
@@ -91,6 +91,11 @@ export interface Lock {
   keyPrice: string
   expirationDuration: number
   key: Key
+  asOf?: number
```
Comment (cnasc) [inline paywall/src/unlockTypes.ts]: Not something that has to be part of this PR, but in the future we may want to have a lock be any of a union type of different lock states, so we can encode when we expect the properties that are optional here to be set

File: paywall/src/unlockTypes.ts (hunk 1 above)
Comment (cellog) [reply paywall/src/unlockTypes.ts]: When the user has no wallet whatsoever. In that case the key is a dummy value used only for display of the lock

File: paywall/src/unlockTypes.ts (hunk 2 above)
Comment (cellog) [reply paywall/src/unlockTypes.ts]: I don't follow (typescript novice), looking forward to picking your brain on this in the near future!

**Pull request merged**